sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample08_EventProcessor.md

```diff
@@ -0,0 +1,230 @@
+# Building a custom processor with EventProcessor
+
+This sample demonstrates using `EventProcessor<TPartition>` to build a custom event processor which manages its own load balancing and checkpoint state and then shows how to use it to consume events from Event Hubs. For most users we strongly recommend using the [EventProcessorClient](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples) from the [Azure.Messaging.EventHubs.Processor](https://www.nuget.org/packages/Azure.Messaging.EventHubs.Processor) package instead of implementing your own processor library.
+
+## What does a processor do?
+
+When consuming events, the consumer is tied to a specific event hub partition and reads the events in order from that partition. This means, to read all events from an Event Hub, you'll want to have one consumer per partition reading and processing events. For some workloads (where the processing can be intensive) you may want these consumers spread out across multiple machines, or even add or remove machines as the rate of incoming events increases or decreases. We call this process *load balancing*. In addition, you'll want to make sure that if one of your consumers has a problem (say the machine it is running on crashes) you're able to resume your processing at a later time, picking up where you left off. We call this process *checkpointing*.
```

Comment:
```suggestion
When consuming events, the consumer is tied to a specific partition of an Event Hub and reads the events in order from that partition. This means, to read all events from an Event Hub, you'll need one consumer per partition reading and processing events. For some workloads (where the processing can be intensive) you may want these consumers spread out across multiple machines, or even to add and remove consumers as the rate of incoming events increases or decreases. We call this process *load balancing*. In addition, you'll want to make sure that if one of your consumers has a problem (such as the machine it is running on crashes) you're able to resume your processing at a later time, picking up where you left off. We call this process *checkpointing*.
```

You may want to consider linking the `load balancing` definition to [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample04_ProcessingEvents.md#load-balancing) write up for more context.   Similarly, there's a blurb on `checkpointing` found [here](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample04_ProcessingEvents.md#checkpointing-while-processing) that may help with additional context.